tiledb/sm/serialization/query.cc

```diff
@@ -271,14 +277,32 @@ Status subarray_partitioner_from_capnp(
     for (size_t i = 0; i < num_attrs; i++) {
       auto mem_budget_reader = mem_budgets_reader[i];
       std::string attr_name = mem_budget_reader.getAttribute();
-      if (attr_name == constants::coords || !schema->var_size(attr_name)) {
-        RETURN_NOT_OK(partitioner->set_result_budget(
-            attr_name.c_str(), mem_budget_reader.getDataBytes()));
+      auto var_size = schema->var_size(attr_name);
+      auto nullable = schema->is_nullable(attr_name);
```

Comment:
Is the var_size/nullable protections good enough here or should these be check if `mem_budget_reader.hasValidityBytes()`? Thinking of backwards compatibility for older libtiledbs.

Reply:
Similar to my other responses, we can't do anything with the validity bytes if they're set but the attribute is nullable. Leaving as-is.